Build error: "export could not write export file"

See:

```
16:30:56 Errors << octomap:cmake /tmp/buildd/workspace/debian/tmp/build/logs-autonomy/octomap/build.cmake.000.log
16:30:56 
16:30:56 
16:30:56 CMake Error: cannot write to file "/tmp/buildd/workspace/octomap-6d7c31ae4df2c93cb8a954e44d442338b58d3558/octomap/lib/cmake/octomap/octomap-targets.cmake": No such file or directory
16:30:56 CMake Error at /tmp/buildd/workspace/octomap-6d7c31ae4df2c93cb8a954e44d442338b58d3558/octomap/src/math/CMakeLists.txt:20 (export):
16:30:56   export could not write export file.
16:30:56 
16:30:56 
16:30:56 CMake Error: cannot write to file "/tmp/buildd/workspace/octomap-6d7c31ae4df2c93cb8a954e44d442338b58d3558/octomap/lib/cmake/octomap/octomap-targets.cmake": No such file or directory
16:30:56 CMake Error at /tmp/buildd/workspace/octomap-6d7c31ae4df2c93cb8a954e44d442338b58d3558/octomap/src/CMakeLists.txt:24 (export):
16:30:56   export could not write export file.
16:30:56 
```

Reverting to before #140 merged resolves the issue.

Looks like this is at least partly down to a bug in old CMake. On Ubuntu Trusty (CMake 2.8.12.2):

```
mkdir src
git clone https://github.com/OctoMap/octomap src/octomap --depth 1
catkin config --install --isolate-devel
catkin build octomap
```

Error is:

```
CMake Error: cannot write to file "<ws>/src/octomap/octomap/lib/cmake/octomap/octomap-targets.cmake": No such file or directory
CMake Error at /home/mpurvis/octo_ws/src/octomap/octomap/src/math/CMakeLists.txt:20 (export):
  export could not write export file.
```

However, doing the exact same under Homebrew (CMake 3.6.2) succeeds just fine, and the following three files are created:

```
./build/octomap/src/CMakeFiles/Export/share/octomap/octomap-targets.cmake
./install/share/octomap/octomap-targets.cmake
./src/octomap/octomap/lib/cmake/octomap/octomap-targets.cmake
```

So first of all, I don't think it's correct behaviour to be generating these into the source folder— they should go in the build tree. But the larger issue is that `export()` on new-CMake creates the folder, which old-CMake doesn't do.

Would we entertain a PR that pre-creates the missing path if it doesn't exist?

--- a/dynamicEDT3D/src/CMakeLists.txt
+++ b/dynamicEDT3D/src/CMakeLists.txt
@@ -14,6 +14,10 @@ target_link_libraries(dynamicedt3d-static ${OCTOMAP_LIBRARIES})
 
 SET_TARGET_PROPERTIES(dynamicedt3d-static PROPERTIES OUTPUT_NAME "dynamicedt3d") 
 
+if(NOT EXISTS "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/cmake/dynamicEDT3D")
+  file(MAKE_DIRECTORY "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/cmake/dynamicEDT3D")
+endif()
+
 export(TARGETS dynamicedt3d dynamicedt3d-static
   FILE "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/cmake/dynamicEDT3D/dynamicEDT3DTargets.cmake")
 

--- a/octomap/src/CMakeLists.txt
+++ b/octomap/src/CMakeLists.txt
@@ -21,6 +21,10 @@ SET_TARGET_PROPERTIES(octomap-static PROPERTIES OUTPUT_NAME "octomap")
 
 TARGET_LINK_LIBRARIES(octomap octomath)
 
+if(NOT EXISTS "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/cmake/octomap")
+  file(MAKE_DIRECTORY "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/cmake/octomap")
+endif()
+
 export(TARGETS octomap octomap-static
   APPEND FILE "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/cmake/octomap/octomap-targets.cmake")
 

--- a/octomap/src/math/CMakeLists.txt
+++ b/octomap/src/math/CMakeLists.txt
@@ -17,6 +17,10 @@ SET_TARGET_PROPERTIES( octomath PROPERTIES
 ADD_LIBRARY( octomath-static STATIC ${octomath_SRCS})
 SET_TARGET_PROPERTIES(octomath-static PROPERTIES OUTPUT_NAME "octomath")
 
+if(NOT EXISTS "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/cmake/octomap")
+  file(MAKE_DIRECTORY "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/cmake/octomap")
+endif()
+
 export(TARGETS octomath octomath-static
   APPEND FILE "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/cmake/octomap/octomap-targets.cmake")
 

--- a/octovis/CMakeLists_src.txt
+++ b/octovis/CMakeLists_src.txt
@@ -113,6 +113,10 @@ IF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
   )
 ENDIF(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
 
+if(NOT EXISTS "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/cmake/octovis")
+  file(MAKE_DIRECTORY "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/cmake/octovis")
+endif()
+
 export(TARGETS octovis octovis-static octovis-shared
   FILE "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}/cmake/octovis/octovis-targets.cmake")
 

